### ISSUE TITLE
tf.losses.mean_squared_error does not support name parameter

### ISSUE THREAD

**Issue by Hong-Xiang** [tensorflow/tensorflow#7939]
Some times MSE is not the only loss in the model, and there maybe multiple MSE losses and adds up to the total loss, a custom name of mse loss layer without using name_scope can be helpful.

**Comment by tatatodd** [comment]
@Hong-Xiang I believe the reason a `name_scope` is used, rather than a single `name` parameter, is because mean_squared_error expands to multiple underlying operations.  Is there a reason why name_scope isn't sufficient for your use case?

**Comment by Hong-Xiang** [comment]
@tatatodd Thank you for your explanation, `name_scope` is sufficient, the only problem is that it may lead a little mess to code than using `name` directly :-).

**Comment by tatatodd** [comment]
@Hong-Xiang OK cool.  And note that although it's a little more code to write, using `name_scope` has the additional benefit that your ops will be nicely grouped in visualization tools like TensorBoard.

I'm closing this out; feel free to file new issues if you run into them!  :)
